ocp_resources/self_node_remediation_templates.py

```diff
@@ -0,0 +1,36 @@
+# API reference: https://github.com/medik8s/self-node-remediation
+from typing import Any
+
+from ocp_resources.resource import NamespacedResource
+
+
+class SelfNodeRemediationTemplate(NamespacedResource):
+    """
+    SelfNodeRemediationTemplate CRD
+    """
+
+    api_group = NamespacedResource.ApiGroup.SELF_NODE_REMEDIATION_MEDIK8S_IO
+
+    def __init__(
+        self,
+        remediation_strategy: str = "",
+        **kwargs: Any,
+    ) -> None:
+        """
+        Create SelfNodeRemediationTemplate object.
+
+        Args:
+            remediation_strategy (str, optional): The remediation method for unhealthy nodes.
+        """
+        super().__init__(
+            **kwargs,
+        )
+        self.remediation_strategy = remediation_strategy
+
+    def to_dict(self) -> None:
+        super().to_dict()
+        if not self.yaml_file:
+            self.res.setdefault("spec", {}).setdefault("template", {}).setdefault("spec", {})
```

Comment:
@myakove Hi, by doing  self.res.setdefault("spec", {}).setdefault("template", {}).setdefault("spec", {}) i am making sure that i have { "spec": { "template": { "spec": {} } } } available because template,spec all are required fields . 

if remediation_strategy is there maybe just adding value self.res["spec"]["template"]["spec"]["remediationStrategy"] = self.remediation_strategy would do instead of direct replacement.

Reply:
you do not need to make sure of it since you set it as literal
`{"template": {"spec": {"remediationStrategy": self.remediation_strategy}}}`

Reply:
self.res.setdefault("spec", {}).setdefault("template", {}).setdefault("spec", {}) is mandatory to have even if remediationStrategy is not set else we will hit this [problem](https://issues.redhat.com/browse/ECOPROJECT-1999)

Reply:
ok, then write it as literal, no need to use `setdefault`